src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+
+        # If a subdir was provided, make sure it exists
+        if repository_subdir:
+            subdir_path = os.path.join(repo_path, repository_subdir)
+            if not os.path.exists(subdir_path):
+                if not interactive:
+                    error("Repository subdirectory does not exist.")
+                logger.info(
+                    f"Subdirectory '{repository_subdir}' does not exist in the "
+                    f"repository."
+                )
+                logger.info("Please enter a valid subdirectory.")
+                repository_subdir = click.prompt(
+                    "Repository subdirectory", default=""
+                )
+                continue
+
+        # Check if the plugin structure is valid.
+        if repository_subdir:
+            plugin_path = os.path.join(repo_path, repository_subdir)
+        else:
+            plugin_path = repo_path
+        try:
+            _validate_repository_structure(plugin_path)
+            return repository_subdir
+        except ValueError as e:
+            msg = (
+                f"Plugin code structure at {repository_subdir} is invalid: "
+                f"{str(e)}"
+            )
+            if not interactive:
+                error(str(e))
+            logger.info(msg)
+            logger.info("Please enter a valid subdirectory.")
+            repository_subdir = click.prompt("Repository subdirectory")
+
+
+def _validate_repository_structure(plugin_root: str) -> None:
+    """Validate the repository structure of a submitted ZenML Hub plugin.
+
+    We expect the following structure:
+    - README.md
+    - requirements.txt
+    - src/
+
+    Args:
+        plugin_root: Root directory of the plugin.
+
+    Raises:
+        ValueError: If the repo does not have the correct structure.
+    """
+    # README.md exists.
+    readme_path = os.path.join(plugin_root, "README.md")
+    if not os.path.exists(readme_path):
+        raise ValueError("README.md not found")
+
+    # requirements.txt exists.
+    requirements_path = os.path.join(plugin_root, "requirements.txt")
+    if not os.path.exists(requirements_path):
+        raise ValueError("requirements.txt not found")
+
+    # src/ exists.
+    src_path = os.path.join(plugin_root, "src")
+    if not os.path.exists(src_path):
+        raise ValueError("src/ not found")
+
+
+def _validate_tags(tags: List[str], interactive: bool) -> List[str]:
+    """Validate the provided tags.
+
+    Args:
+        tags: The tags to validate.
+        interactive: Whether to run in interactive mode.
+
+    Returns:
+        The validated tags.
+    """
+    if not tags:
+        if not interactive:
+            return []
+
+        # In interactive mode, ask for tags if none were provided.
+        return _ask_for_tags()
+
+    # If tags were provided, print a warning if any of them is invalid.
+    for tag in tags:
+        if tag.startswith(ZENML_HUB_INTERNAL_TAG_PREFIX):
```

Comment:
I'm assuming that this check also exists serverside to make sure anyone modifying/bypassing our CLI can't use that tag?

Reply:
yep we have the same checks server-side 👍